src/nvim/terminal.c

```diff
@@ -333,8 +333,8 @@ void terminal_enter(bool process_deferred)
   int save_rd = RedrawingDisabled;
   State = TERM_FOCUS;
   RedrawingDisabled = false;
-  bool save_mapped_ctrl_c = mapped_ctrl_c;
-  mapped_ctrl_c = true;
+  int save_mapped_ctrl_c = mapped_ctrl_c;
+  mapped_ctrl_c = MAP_ALL_MODES;
```

Comment:
Before, `mapped_ctrl_c` was a Boolean value, where `true` indicated that Ctrl-C was mapped and `false` that it wasn't mapped. Now that `mapped_ctrl_c` is an `int` with mode flags, the `true` case becomes 'all flags turned on', which I think can be expressed as `MAP_ALL_MODES`. Correct?

(Why is Ctrl-C always 'mapped' when entering the terminal though? I don't understand this.)